payu/subcommands/collate_cmd.py

```diff
@@ -23,10 +23,30 @@ def runcmd(model_type, config_path, init_run, n_runs, lab_path, dir_path):
     pbs_config = cli.get_config(config_path)
     pbs_vars = cli.set_env_vars(init_run, n_runs, lab_path, dir_path)
 
-    collate_queue = pbs_config.get('collate_queue', 'copyq')
+    collate_conf = pbs_config.get('collate',{})
+
+    # Cycle through old collate config and convert to newer dict format
+    if type(collate_conf) is bool:
+        collate_conf = {'run' : collate_conf}
+        for key in ['queue','ncpus','walltime','mem']:
+            oldkey = "collate_{}".format(key)
+            print("Use of this key is deprecated: {}. Use collate dictionary and subkey {}".format(oldkey,key))
+            if oldkey in pbs_config:
+                collate_conf[key] = pbs_config[oldkey]
```

Comment:
I was going to again say "move this to `read_config`, but I notice that `cli.get_config` does not use `read_config`, even though it's extremely similar.

Might be worth merging these functions somehow.

Reply:
The major difference seems to be in their error handling. The `cli` version is willing to have config_path as `None`. Is this important?
```python
except (TypeError, IOError) as exc:
        if config_path is None:
            config = {}
        elif type(exc) == IOError and exc.errno == errno.ENOENT:
            print('payu: error: Configuration file {} not found.'
                  ''.format(config_path))
            sys.exit(errno.ENOENT)
        else:
            raise

    return config```

Reply:
This may not have even been tested... I don't think I've run anything with no `config.yaml`... 

Using `payu.fsops.read_config` seems to "work" on some basic cases, so maybe let's just try replacing `get_config` with `read_config` in `collate_cmd.py` and see how it goes?

Reply:
I assume I'll replace it wherever it appears:
```
$ grep -ir get_config payu 
payu/subcommands/run_cmd.py:    pbs_config = cli.get_config(config_path)
payu/subcommands/profile_cmd.py:    pbs_config = cli.get_config(config_path)
payu/subcommands/collate_cmd.py:    pbs_config = cli.get_config(config_path)
```

Reply:
Going to open an issue about `read_config` and `get_config`; there's more I can say but it's becoming a distraction.  For now, let's exclusively use `read_config` and phase out `get_config`, starting with `collate_cmd.py`.

Reply:
Oops. I just replaced them all.